Source/plugins/Types.h

```
@@ -59,44 +59,50 @@ namespace PluginHost {
             }
             void Register(IShell* controller, const string& callsign)
             {
-                _adminLock.Lock();
-                _callsign = callsign;
-                _state = state::REGISTRING;
-                _adminLock.Unlock();
-
-                controller->Register(this);
-
-                _adminLock.Lock();
-                if (_state == state::LOADED) {
-
-                    INTERFACE* entry = _designated->QueryInterface<INTERFACE>();
-                    _designated->Release();
-                    _designated = entry;
-                    _state = state::RUNNING;
-                    if (entry != nullptr) {
-                        _parent.Activated(entry);
+                ASSERT(controller != nullptr);
```

Here I think the addition of the ASSERT should be sufficient. Even in Release, let the application segfault. We are now leaving (igf this is called with a controller that is a nullptr, the object in a undefined state that might lead to unexpected crashes later on. I prefer the crash to happen now. So agree on the addition of the ASSERT but lets not try to work around it. Let the application crash. It is up to the callee to make sure the controller parameter is *not* null.

---

done